libs/vkd3d-shader/vkd3d_shader_main.c

```diff
@@ -300,6 +302,29 @@ static void vkd3d_shader_scan_destroy(struct vkd3d_shader_scan_info *scan_info)
     hash_map_clear(&scan_info->register_map);
 }
 
+static int vkd3d_shader_validate_shader_type(enum vkd3d_shader_type type, VkShaderStageFlagBits stages)
+{
+    static const VkShaderStageFlagBits table[VKD3D_SHADER_TYPE_COUNT] = {
+        VK_SHADER_STAGE_FRAGMENT_BIT,
+        VK_SHADER_STAGE_VERTEX_BIT,
+        VK_SHADER_STAGE_GEOMETRY_BIT,
+        VK_SHADER_STAGE_TESSELLATION_CONTROL_BIT,
+        VK_SHADER_STAGE_TESSELLATION_EVALUATION_BIT,
+        VK_SHADER_STAGE_COMPUTE_BIT,
+    };
+
+    if (type >= VKD3D_SHADER_TYPE_COUNT)
+        return VKD3D_ERROR_INVALID_ARGUMENT;
+
+    if (!(table[type] & stages))
```

Comment:
Is the intention here that `stages` can contain multiple bits or just one? Should either fix the type (to `VkShaderStageFlags`) or make this `!=` instead.

Reply:
Can be STAGE_ALL if every stage is acceptable (mostly for RT)

Reply:
I guess it's not that relevant since I don't let RT paths go through this check anyways, so should probably just be !=

Reply:
Fixed.